File: java/Sort/SelectSort.java
import java.util.*;
public class SelectSort {

    public static void Select_Sort(int[] nums) {
        for (int i = 0; i < nums.length - 1; i++) {
            int minIndex = i; 
            for (int j = i + 1; j < nums.length; j++) {
                if (nums[minIndex] > nums[j]) {
                    minIndex = j; 
                }
            }
            
            if (i != minIndex) {
                int temp = nums[minIndex];
                nums[minIndex] = nums[i];
                nums[i] = temp;
            }
        }
       
        for (int num : nums) {
            System.out.print(num + " ");
        }
    }

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int[] nums = new int[n];
        String[] str = new String[n];
        for(int i = 0; i < n; i++){
            str[i] = in.next();
            nums[i] = Integer.parseInt(str[i]);
        }

        for(int i = 0; i < n; i++){

        }
        Select_Sort(nums);
        in.close();
    }
}

File: c/Sort/SelectSort.cpp
#include "stdio.h"

void swap(int a, int b){
    int temp;
    temp = a;
    a = b;
    b = temp;
}

void printSc(int nums[], int n){
    int i;
    for(i = 0; i < n; i++){
        printf("%d", nums[i]);
    }
}

void selectSort(int nums[], int n){
    int i, j;
    for(i = 0; i < n - 1; i++){
        int min = i;
        for(j = i + 1; j < n; j++){
            if(nums[j] < nums[min]){
                min = j;
            }
        }
        swap(nums[min], nums[i]);
    }
    printSc(nums, n);
}


int main(){
    int i, n;
    scanf("%d", &n);
    int nums[1000000];
    for(i = 0; i < n; i++){
        scanf("%d", &nums[i]);
    }
    for(i = 0; i < n; i++){
        printf("%d", &nums[i]);
    }
}
